nss/src/client/mod.rs

```diff
@@ -0,0 +1,18 @@
+use authd::nss_client::NssClient;
+use std::error::Error;
+use tokio::net::UnixStream;
+use tonic::transport::{Channel, Endpoint, Uri};
+use tower::service_fn;
+
+pub mod authd {
+    tonic::include_proto!("authd");
+}
+
+pub async fn new_client() -> Result<NssClient<Channel>, Box<dyn Error>> {
+    // The URL is not used, but it must be valid.
+    let channel = Endpoint::try_from("http://[::]:50051")?
+        .connect_with_connector(service_fn(|_: Uri| UnixStream::connect(super::SOCKET_PATH)))
```

Comment:
Do you mind setting that up now, so that we can use it in our testing environment without using the hardcoded socket from the get go? (I plan to do the daemon part next pulse).